### PR TITLE
NullPointerException in InstructorFeedbackQuestionEditAction #4705

### LINKED ISSUE #4705: NullPointerException in InstructorFeedbackQuestionEditAction
java.lang.NullPointerException
	

Actual user abc
Request Method POST 
User Agent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10 
Request Url https://teammatesv4.appspot.com/page/instructorFeedbackQuestionEdit 
Request Path /page 
Request Parameters {showresponsesto::RECEIVER,OWN_TEAM_MEMBERS,RECEIVER_TEAM_MEMBERS,INSTRUCTORS, showrecipientto::RECEIVER,INSTRUCTORS, numofrecipients::1, questionedittype::edit, questiontext::Using the Situation-Behavior-Impact method of feedback discussed in class, provide one statement regarding the student's contribution to date., fsname::Feedback assignment #1, questionid::abczfnRlYW1tYXRlc3Y0ch0LEhBGZWVkYmFja1F1ZXN0aW9uGICAgPbkoI0KDA, numofrecipientstype::max, recipienttype::OWN_TEAM_MEMBERS_INCLUDING_SELF, showgiverto::INSTRUCTORS, receiverLeaderCheckbox::RECEIVER, questiontype::CONTRIB, questionnum::1//1, givertype::STUDENTS, user::abc, courseid::MGMT210-spring2016} 
Stack Trace

java.lang.NullPointerException
	at teammates.logic.core.FeedbackQuestionsLogic.updateFeedbackQuestionNumber(FeedbackQuestionsLogic.java:552)
	at teammates.logic.api.Logic.updateFeedbackQuestionNumber(Logic.java:1810)
	at teammates.ui.controller.InstructorFeedbackQuestionEditAction.editQuestion(InstructorFeedbackQuestionEditAction.java:97)
	at teammates.ui.controller.InstructorFeedbackQuestionEditAction.execute(InstructorFeedbackQuestionEditAction.java:49)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:337)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:59)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:143)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:260)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:78)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:148)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:468)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:256)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:235)
	at java.lang.Thread.run(Thread.java:745)

### PR DESCRIPTION
**Pull request by unyoungwax**
Fixes #4705

A check is added when updating question number (similar to `updateFeedbackQuestion`).

### CODE REVIEW

**Comment by kanghj** [comment]
@unyoungwax, can you add a test case for this?

**Comment by thyageshm** [comment]
@unyoungwax could we try and finish this today?

**Comment by unyoungwax** [comment]
I have added a new test case to check if editing a non-existent question will result in the correct error page.

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java [hunk 1]
```diff
@@ -107,7 +107,8 @@ private void testGeneralQuestionOperations() throws Exception {
         testAjaxOnVisibilityMessageButton();
 
         testDeleteQuestionAction(2);
-        testDeleteQuestionAction(1);
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: Will it be possible to add the test without removing an existing test case? Even if the old test case isn't useful, I think it's preferable if they are not removed without good reason in this pr.

File: src/main/java/teammates/logic/backdoor/BackDoorServlet.java [hunk 2]
```diff
@@ -261,8 +262,11 @@ private String executeBackendAction(HttpServletRequest req, String action)
             return backDoorLogic.getFeedbackQuestionAsJson(feedbackSessionName, courseId, qnNumber);            
         }  else if (action.equals(OPERATION_GET_FEEDBACK_QUESTION_FOR_ID_AS_JSON)) { 
             String questionId = req.getParameter(PARAMETER_FEEDBACK_QUESTION_ID);
-            return backDoorLogic.getFeedbackQuestionForIdAsJson(questionId);            
-        }  else if (action.equals(OPERATION_GET_FEEDBACK_RESPONSE_AS_JSON)) { 
+            return backDoorLogic.getFeedbackQuestionForIdAsJson(questionId);
+        } else if (action.equals(OPERATION_DELETE_FEEDBACK_QUESTION)) {
```
Comment (kanghj) [inline src/main/java/teammates/logic/backdoor/BackDoorServlet.java]: You will have to add a test case for this in BackDoorTest, possibly in a separate issue.

**Comment by unyoungwax** [comment]
I have modified the new test case so that it does not remove existing test cases.

**Comment by kanghj** [comment]
Looks fine to me. If you are not going to write the test case for the new backdoor operation here, can you make a new issue and work on it? 

@damithc, this looks ready for your review.

**Comment by unyoungwax** [comment]
I will create a new issue after this pr is merged.

**Comment by kanghj** [comment]
@unyoungwax, remember to make the new issue

**Pull request merged**